src/gretel_client/pkg_installers.py

```diff
@@ -1,45 +1,78 @@
+import selectors
 import sys
 import subprocess
 
 import requests
 
 
-PKG_ENDPOINT = 'https://{}/opt/pkg'
+PKG_ENDPOINT = "https://{}/opt/pkg"
 
-TX_PKG = 'gretel-transformers'
+TX_PKG = "gretel-transformers"
 
 
 class GretelInstallError(Exception):
     pass
 
 
-def _install_pip_dependency(dep: str):
-    """Install pip dependency via a subprocess"""
-    cmd = [sys.executable, "-m", "pip", "install", dep]
-    print(f"running: {' '.join(cmd)}")
-    results = subprocess.Popen(cmd, stdout=subprocess.PIPE,
-                               stderr=subprocess.STDOUT)
-    for line in iter(results.stdout.readline, b''):  # pragma: no cover
+def read_pipe(section, cmd, p, verbose, collect=False):
+    out = [
+        f"\n{section}\n{''.join(['=' for _ in range(len(section)+1)])}",
+        " ".join(cmd),
+    ]
+    if verbose and not collect:
+        for line in out:
+            print(line)
+    for line in iter(p.readline, b""):
         if isinstance(line, bytes):
-            print(line.decode('utf-8').strip())  # type: ignore
-        else:
+            line = line.decode("utf-8").strip()  # type: ignore
+        if verbose and not collect:
             print(line)
+        if collect:
+            out.append(line)
+
+    if verbose and collect and len(out) > 2:
+        for line in out:
+            print(line)
+        print("")
+
+
+def _install_pip_dependency(dep: str, verbose: bool = True):
+    """Install pip dependency via a subprocess"""
+    cmd = [
+        sys.executable,
+        "-m",
+        "pip",
+        "--disable-pip-version-check",
+        "install",
+        dep,
+    ]
+    if verbose:
+        print(f"running: {' '.join(cmd)}")
+    results = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    read_pipe("pip: Install Output", cmd, results.stdout, verbose)
+    read_pipe("pip: Errors & Warnings", cmd, results.stderr, verbose=True, collect=True)
 
 
 def _get_package_endpoint(package_identifier: str, api_key: str, host: str) -> str:
     """Retrieve package installation endpoints from Gretel API"""
-    pkg_resp = requests.get(f'{PKG_ENDPOINT.format(host)}/{package_identifier}',
-                            headers={'Authorization': api_key})
+    pkg_resp = requests.get(
+        f"{PKG_ENDPOINT.format(host)}/{package_identifier}",
+        headers={"Authorization": api_key},
+    )
 
     if pkg_resp.status_code != 200:  # pragma: no cover
-        raise GretelInstallError('Could not fetch package details from '
-                                 'endpoint.')
+        raise GretelInstallError("Could not fetch package details from " "endpoint.")
 
     details = pkg_resp.json()
-    source_pkg = details['data']['package']['wheel']
+    source_pkg = details["data"]["package"]["wheel"]
     return source_pkg
 
 
-def install_transformers(api_key: str, host: str):
+def install_transformers(api_key: str, host: str, verbose: bool = False):
```

Comment:
Can we make this `install_packages` or something so it's not confusing / pigeonholed to transformers? I'd create an alias `install_transformers` too since we do have that dependency, but just mark it for eventual deprecation

Reply:
Updated. `install_transformers()` will emit a `warning` if used.